### PR TITLE
feat: migrate namespace to TIM

### PR DESCRIPTION
**Pull request by iamar7**
### Description
Migration of namespace module to public

https://github.ibm.com/GoldenEye/issues/issues/6992
### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [x] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [x] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,52 +38,76 @@ https://terraform-ibm-modules.github.io/documentation/#/implementation-guideline
...
+  host                   = data.ibm_container_cluster_config.cluster_config.host
+  token                  = data.ibm_container_cluster_config.cluster_config.token
+  cluster_ca_certificate = data.ibm_container_cluster_config.cluster_config.ca_certificate
+}
+
+##############################################################################
+# Namespace Module
+##############################################################################
+
+# Replace "main" with a GIT release version to lock into a specific release
```
Comment (ocofaigh) [inline README.md:72]: remove this comment

File: README.md [hunk 2]
```diff
@@ -35,52 +38,76 @@ https://terraform-ibm-modules.github.io/documentation/#/implementation-guideline
...
+module "namespace" {
+  source            = "terraform-ibm-modules/namespace-module/ibm"
```
Comment (ocofaigh) [inline README.md:74]: repo was renamed, so update this

File: examples/basic/README.md [hunk 3]
```diff
@@ -1,11 +1,11 @@
-# Basic example
+# Basic Example
```
Comment (ocofaigh) [inline examples/basic/README.md:1]: Basic Example -> Basic example

File: examples/basic/main.tf [hunk 4]
```diff
@@ -1,24 +1,111 @@
...
+
+
+##############################################################################
+# Resource Group
+##############################################################################
 
 module "resource_group" {
-  source  = "terraform-ibm-modules/resource-group/ibm"
-  version = "1.1.4"
+  source = "git::https://github.com/terraform-ibm-modules/terraform-ibm-resource-group.git?ref=v1.1.4"
```
Comment (ocofaigh) [inline examples/basic/main.tf:15]: please use `terraform-ibm-modules/resource-group/ibm`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
a few small comments

File: .github/settings.yml [hunk 5]
```diff
@@ -22,7 +22,7 @@ repository:
 
   # Uncomment this description property
   # and update the description to the current repo description.
-  # description: ""
+  description: "Configures a Kubernetes namespace or Openshift project."
 
   # Use a comma-separated list of topics to set on the repo (ensure not to use any caps in the topic string).
   topics: terraform, ibm-cloud, terraform-module
```
Comment (ocofaigh) [inline .github/settings.yml:28]: please add these topics too: core-team, supported, graduated, namspace, project, kubernetes, openshift

**Review by ocofaigh — CHANGES_REQUESTED** [review]
just 1 comment to update topics

File: .github/settings.yml (hunk 5 above)
Comment (iamar7) [reply .github/settings.yml]: Updated the topics.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**